src/http.rs

```diff
@@ -62,18 +73,41 @@ fn confidence(block_num: u64, db: Arc<DB>) -> ClientResponse<ConfidenceResponse>
 				serialised_confidence,
 			})
 		},
-		Ok(None) => ClientResponse::NotFound,
+		Ok(None) => {
+			let lock = counter.lock().unwrap();
+			if block_num < *lock {
+				return ClientResponse::NotFinalized;
+			} else {
+				return ClientResponse::NotFound;
+			}
+		},
 		Err(e) => ClientResponse::Error(e),
 	};
 	info!("Returning confidence: {res:?}");
 	res
 }
 
+fn latest_block(counter: Arc<Mutex<u64>>) -> ClientResponse<LatestBlockResponse> {
+	info!("Got request for latest block");
+	let res = match counter.lock() {
+		Ok(counter) => ClientResponse::Normal(LatestBlockResponse {
+			latest_block: *counter,
+		}),
+		Err(_) => ClientResponse::NotFound,
+	};
+	res
+}
+
 fn appdata(
 	block_num: u64,
 	db: Arc<DB>,
 	cfg: &RuntimeConfig,
+	counter: Arc<Mutex<u64>>,
 ) -> ClientResponse<ExtrinsicsDataResponse> {
+	let lock = counter.lock().unwrap();
```

Comment:
Should we use `unwrap` here? Or some other mean of error handling? Don't know what are the consequences of not being able to acquire the lock here...